src/tools/rust-analyzer/crates/proc-macro-api/src/version.rs

```diff
@@ -120,17 +120,20 @@ pub fn read_version(dylib_path: &AbsPath) -> io::Result<String> {
     let version = u32::from_be_bytes([dot_rustc[4], dot_rustc[5], dot_rustc[6], dot_rustc[7]]);
     // Last supported version is:
     // https://github.com/rust-lang/rust/commit/0696e79f2740ad89309269b460579e548a5cd632
-    match version {
-        5 | 6 => {}
+    let snappy_portion = match version {
+        5 | 6 => &dot_rustc[8..],
+        7 => {
+            let len_bytes = &dot_rustc[8..12];
+            let data_len = u32::from_be_bytes(len_bytes.try_into().unwrap()) as usize;
+            &dot_rustc[12..data_len + 12]
+        },
```

Comment:
Does `rustc` use a nightly `rustfmt`? As tiny as this is, I feel this is going to cause us a lot of pain (cf https://github.com/rust-lang/rust-analyzer/pull/14153).

Reply:
Yes, rustc uses a nightly rustfmt. Rust-analyzer shouldn't be formatted though: https://github.com/rust-lang/rust/blob/9bb6e60d1f1360234aae90c97964c0fa5524f141/rustfmt.toml#L35

Reply:
```suggestion
        }
```

Reply:
Note that r-a will happily format this since rustfmt doesn't know about file paths when invoked by r-a currently

Reply:
I can remove the comma here to hopefully sidestep conflicts, if desired. Or rebase, or try to pull the commits I made to r-a in here? I'm not super familiar with git subtrees like this.

Reply:
> Note that r-a will happily format this since rustfmt doesn't know about file paths when invoked by r-a currently

Great, another reason for me to be anxious on subtree syncs 😵‍💫.

I think you can drop it here in a new commit (and optionally squash them).